scipy/optimize/direct.py

```diff
@@ -0,0 +1,259 @@
+    direct = None
+
+
+__version_info__ = ('1', '1')
+__version__ = '.'.join(__version_info__)
+
+ERROR_MESSAGES = (
+    'u[i] < l[i] for some i',
+    'maxf is too large',
+    'Initialization failed',
+    'There was an error in the creation of the sample points',
+    'An error occured while the function was sampled',
+    'Maximum number of levels has been reached.',
+)
+
+SUCCESS_MESSAGES = (
+    'Number of function evaluations done is larger then maxf',
+    'Number of iterations is equal to maxT',
+    'The best function value found is within fglper of the (known) global optimum',
+    'The volume of the hyperrectangle with best function value found is below volper',
+    'The volume of the hyperrectangle with best function value found is smaller then volper'
+)
+
+# Class for returning the result of an optimization algorithm (copied from
+# scipy.optimize)
+class OptimizeResult(dict):
+    r""" Represents the optimization result.
+
+    Attributes
+    ----------
+    x : ndarray
+        The solution of the optimization.
+    success : bool
+        Whether or not the optimizer exited successfully.
+    status : int
+        Termination status of the optimizer. Its value depends on the
+        underlying solver. Refer to `message` for details.
+    message : str
+        Description of the cause of the termination.
+    fun, jac, hess, hess_inv : ndarray
+        Values of objective function, Jacobian, Hessian or its inverse (if
+        available). The Hessians may be approximations, see the documentation
+        of the function in question.
+    nfev, njev, nhev : int
+        Number of evaluations of the objective functions and of its
+        Jacobian and Hessian.
+    nit : int
+        Number of iterations performed by the optimizer.
+    maxcv : float
+        The maximum constraint violation.
+
+    Notes
+    -----
+    There may be additional attributes not listed above depending of the
+    specific solver. Since this class is essentially a subclass of dict
+    with attribute accessors, one can see which attributes are available
+    using the `keys()` method.
+    """
+    def __getattr__(self, name):
+        try:
+            return self[name]
+        except KeyError:
+            raise AttributeError(name)
+
+    __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+    def __repr__(self):
+        if self.keys():
+            m = max(map(len, list(self.keys()))) + 1
+            return '\n'.join([k.rjust(m) + ': ' + repr(v)
+                              for k, v in self.items()])
+        else:
+            return self.__class__.__name__ + "()"
+
+def _minimize_direct(func, bounds=None, nvar=None, args=(), disp=False,
+             eps=1e-4,
+             maxf=20000,
+             maxT=6000,
+             algmethod=0,
+             fglobal=-1e100,
+             fglper=0.01,
+             volper=-1.0,
+             sigmaper=-1.0,
+             **kwargs
```

Comment:
`kwargs` isn't used, would be good to remove, unless it's needed to make `minimize()` happy.